src/execution/as_execution.rs

```diff
@@ -24,25 +24,27 @@ impl MassaModule for ASModule {
     fn get_bytecode(&self) -> &Vec<u8> {
         &self.bytecode
     }
-    fn execution(&self, instance: &Instance, function: &str, param: &[u8]) -> Result<Response> {
+    fn execution(&self, instance: &Instance, store: &mut Store, function: &str, param: &[u8]) -> Result<Response> {
         if cfg!(not(feature = "gas_calibration")) {
             // sub initial metering cost
             let metering_initial_cost = settings::metering_initial_cost();
-            let remaining_gas = get_remaining_points(&self.env)?;
+            let remaining_gas = get_remaining_points(&self.env, store)?;
             if metering_initial_cost > remaining_gas {
                 bail!("Not enough gas to launch the virtual machine")
             }
-            set_remaining_points(&self.env, remaining_gas - metering_initial_cost)?;
+            set_remaining_points(&self.env, store, remaining_gas - metering_initial_cost)?;
         }
 
         // Now can exec
         let wasm_func = instance.exports.get_function(function)?;
-        let argc = wasm_func.param_arity();
+        let argc = wasm_func.param_arity(store);
         let res = if argc == 0 && function == crate::settings::MAIN {
-            wasm_func.call(&[])
+            wasm_func.call(store, &[])
         } else if argc == 1 {
-            let param_ptr = *BufferPtr::alloc(&param.to_vec(), self.env.get_wasm_env())?;
-            wasm_func.call(&[Val::I32(param_ptr.offset() as i32)])
+            // TODO: no unwrap() & need env in ::alloc()?
+            let memory = self.env.get_wasm_env().memory.as_ref().unwrap();
+            let param_ptr = *BufferPtr::alloc(&param.to_vec(), self.env.get_wasm_env(), memory, store)?;
```

Comment:
It seems indeed that `write_buffer` (function in the bindings that requires you to send `env` to `alloc` in the first place) never uses this parameter. This goes together with my above comment, solution might be to use only memory, but having only a ref to the env as unified input would be justified too, in any case having both makes no sense to me

Reply:
done